pywbem/_logging.py

```diff
@@ -32,53 +32,103 @@
+Logging is activated in cim_operations.WBEMConnection with the `enable_logging`
+parameter.
 
-* :meth:`~pywbem._logging.PywbemLoggers.create_logger` creates one or more
-  loggers from
-  parameter inputs that define the component name and characteristics of each
-  logger.
+The following example sets up the operations logger using Python logger
+methods to log to stdout::
 
-These functions save the logger definitions in the
-:class:`~pywbem._logging.PywbemLoggers` class that
-is used by the logging functions in the recorder so trying to create the named
-loggers independently of this code may cause issues.
+    logger = logging.getLogger('pywbem.ops')
+    handler = logging.StreamHandler()
+    handler.setLevel(logging.DEBUG)
 
-The pywbem loggers are based on two parameters (log destination, log detail)
+    formatter = logging.Formatter('%(asctime)s - %(name)s - %(message)s')
+    handler.setFormatter(formatter)
+    logger.addHandler(handler)
+
+    conn.WBEMConnection(... enable_log='all')
+
+When the debug level is set, the pywbem logger will generate log a log entry
+for each WBEMConnection request and response.
+
+The helper functions in _logging are only intended to provide tools for
+users of pywbem that want simply one or two line setup for logs as follows:
+
+* :meth:`~pywbem._logging.define_loggers_from_string` defines one or more
+  loggers from  a string input that defines the component name and
+  characteristics of each logger.  This allows other tools like CLIs that use
+  pywbem to create the pywbem known logs with minimal work from command line or
+  config file input.
+
+* :meth:`~pywbem._logging.define_logger` defines  one
+  logger from parameter inputs that define the component name and
+  characteristics of the logger.
+
+Thus to setup loggers that log all of the operations requests and responses
+both for http and the APIs either of the following could be use::
+
+    # Log ops to stderr and http to file but only log the first n characters
+    # where n is defined by the pywbem variable `DEFAULT_MAX_LOG_ENTRY_SIZE`
+    define_logger('ops', 'stderr')
+    define_logger('http', 'file', log_filename='mylog.log')
+    conn = WBEMConnection(..., enable_log=min)
+
+    # Log everything to stderr and output all of the information
+    define_loggers_from_string('all=stderr')
+    conn = WBEMConnection(..., enable_log='all')
+
+    # Log only http requests and responses to a file
+    define_logger('http', 'file', log_filename='mylog.log')
+    conn = WBEMConnection(..., enable_log=min)
+
+The pywbem loggers are based one parameters (log destination)
 that determine if the logs for the logger name are created,
 how much information is inserted, and the log destination.  This
 extends the python logging facility to include the log_detail parameter which
 defines whether all the information defined or only a limited size is output.
 This is used with pywbem because the logs on operation and http responses can be
 very large.
```

Comment:
The previous paragraph needs to be rewritten - we are no longer extending the Python logging facility by a log_detail parameter.

Reply:
Cleaned up